File: Java/src/main/java/planes/ExperimentalPlane.java
package planes;

import models.ClassificationLevel;
import models.ExperimentalTypes;

public class ExperimentalPlane extends Plane {

    private ExperimentalTypes type;
    private ClassificationLevel classificationLevel;

    public ExperimentalPlane(String model, int maxSpeed, int maxFlightDistance, int maxLoadCapacity, ExperimentalTypes type, ClassificationLevel classificationLevel) {
        super(model, maxSpeed, maxFlightDistance, maxLoadCapacity);
        this.type = type;
        this.classificationLevel = classificationLevel;
    }

    public ClassificationLevel getClassificationLevel(){

        return classificationLevel;
    }

    public void setClassificationLevel(ClassificationLevel classificationLevel) {
        this.classificationLevel = classificationLevel;
    }
    @Override
    public String toString() {
        return "experimentalPlane{" +
                "model='" + model + '\'' +
                '}';
    }
}

File: JS/Planes/experimentalPlane.js
const Plane = require('./Plane');

class ExperimentalPlane extends Plane {
  constructor(
    model,
    maxSpeed,
    maxFlightDistance,
    maxLoadCapacity,
    type,
    classificationLevel
  ) {
    super(model, maxSpeed, maxFlightDistance, maxLoadCapacity);
    this.model = model;
    this.maxSpeed = maxSpeed;
    this.maxFlightDistance = maxFlightDistance;
    this.maxLoadCapacity = maxLoadCapacity;
    this.type = type;
    this.classificationLevel = classificationLevel;
  }

  get model() {
    return this.model;
  }

  set model(value) {
    this.model = value;
  }

  get maxSpeed() {
    return this.maxSpeed;
  }

  set maxSpeed(value) {
    this.maxSpeed = value;
  }

  get maxFlightDistance() {
    return this.maxFlightDistance;
  }

  set maxFlightDistance(value) {
    this.maxFlightDistance = value;
  }

  get maxLoadCapacity() {
    return this.maxLoadCapacity;
  }

  set maxLoadCapacity(value) {
    this.maxLoadCapacity = value;
  }

  get type() {
    return this.type;
  }

  set type(value) {
    this.type = value;
  }

  get classificationLevel() {
    return this.classificationLevel;
  }

  set classificationLevel(value) {
    this.classificationLevel = value;
  }
}

module.exports = ExperimentalPlane;
